tests/integrationv2/test_happy_path.py

```diff
@@ -57,7 +58,14 @@ def test_s2n_server_happy_path(managed_process, cipher, provider, curve, protoco
     # the stdout reliably.
     for results in server.get_results():
         assert results.exception is None
-        assert results.exit_code == 0
+        '''
+        In TLS1.3 the JDK sends a user_canceled alert as well as a close_notify alert
+        after a connection closes, so s2n will register the alert and complain.
+        '''
+        if protocol.name == "TLS1.3" and provider is JavaSSL:
```

Comment:
In the JDK are the alerts throwing exceptions? Can we catch those exceptions and treat them as expected in the JDK client (preventing modification of this test) ?

Reply:
The jdk client is not erroring here. It is sending closure two alerts when its connection closes, which is a proper way of interpreting the rfc. It is s2n which is erroring, because I think this is something we missed when writing TLS1.3 behavior. (https://tools.ietf.org/html/rfc8446#section-6.1) user_canceled is a closure alert, not an error alert.